src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -246,6 +352,60 @@ public void resetNeededResources()
     }
 
     /**
+     * Update the available resources (from the chest or builder's inventory)
+     * which are needed for the build
+     */
+    private void updateAvailableResources()
+    {
+        final EntityCitizen builder = getWorkerEntity();
+        resourcesAvailable.clear();
+
+
+        InventoryCitizen builderInventory = null;
+        if (builder!=null)
+        {
+            builderInventory = builder.getInventoryCitizen();
+        }
+
+
+        for (@NotNull final Map.Entry<String, ItemStack> entry : neededResources.entrySet())
+        {
+            ItemStack itemStack = entry.getValue();
+            BuildingBuilderResource resource = resourcesAvailable.get(entry.getKey());
+            if (resource == null)
+            {
+                resource = new BuildingBuilderResource(entry.getValue().getDisplayName(), itemStack,0,itemStack.getCount());
+            }
+
+            if (builderInventory!=null)
+            {
+                resource.setAvailable(resource.getAvailable() + InventoryUtils.getItemCountInInventory(builderInventory, entry.getValue().getItem(), entry.getValue().getItemDamage()));
+            }
+
+            final IInventory chestInventory = this.getTileEntity();
+            if (chestInventory!=null)
+            {
+                resource.setAvailable(resource.getAvailable() + InventoryUtils.getItemCountInInventory(chestInventory, entry.getValue().getItem(), entry.getValue().getItemDamage()));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Split this 182 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)